packages/mobile/src/identity/feelessVerification.ts

```diff
@@ -0,0 +1,1061 @@
+
+  let { phoneHashDetails } = feelessVerificationState
+  let { phoneHash } = feelessVerificationState.phoneHashDetails
+  const ownPepper = pepperCache[e164Number]
+
+  // If we already know the phoneHash, we can bail without updating state
+  if (phoneHash.length) {
+    return phoneHashDetails
+  }
+
+  // Pepper will be queried as part of feeless verification flow but cache
+  // may be empty if the user hasn't initiated a Komenci session before
+  if (!ownPepper) {
+    Logger.debug(TAG, '@fetchPhoneHashDetailsFromCache', 'Pepper not cached')
+    throw new PepperNotCachedError()
+  }
+
+  phoneHash = getPhoneHash(e164Number, ownPepper)
+  phoneHashDetails = { e164Number, phoneHash, pepper: ownPepper }
+
+  yield put(
+    feelessUpdateVerificationState({
+      ...feelessVerificationState,
+      phoneHashDetails,
+    })
+  )
+  return phoneHashDetails
+}
+
+function* fetchSessionState(komenciKit: KomenciKit, e164Number: string) {
+  Logger.debug(TAG, '@fetchSessionState', 'Starting fetch')
+  const [feelessVerificationState, pepperCache, sessionStatusResult]: [
+    FeelessVerificationState,
+    E164NumberToSaltType,
+    Result<CheckSessionResp, FetchError>
+  ] = yield all([
+    select(feelessVerificationStateSelector),
+    select(e164NumberToSaltSelector),
+    call([komenciKit, komenciKit.checkSession]),
+  ])
+
+  let sessionActive = true
+  let {
+    requestQuotaRemaining,
+    pepperQuotaRemaining,
+    unverifiedMtwAddress,
+  } = feelessVerificationState.komenci
+
+  // An inactive session is not fatal, it just means we will need to start one
+  if (!sessionStatusResult.ok) {
+    Logger.debug(TAG, '@fetchSessionState', 'No active sessions')
+    sessionActive = false
+  } else {
+    Logger.debug(TAG, '@fetchSessionState', 'Active session found')
+    const {
+      quotaLeft: { distributedBlindedPepper, requestSubsidisedAttestation, submitMetaTransaction },
+      metaTxWalletAddress,
+    } = sessionStatusResult.result
+    const ownPepper = pepperCache[e164Number]
+
+    Logger.debug(
+      TAG,
+      '@fetchSessionState Session status:',
+      JSON.stringify(sessionStatusResult.result)
+    )
+
+    sessionActive = true
+    // Somtimes `metaTxWalletAddress` is returned as undefined for an active session.
+    // In that case, use the `unverifiedMtwAddress` we have stored locally
+    unverifiedMtwAddress = metaTxWalletAddress ?? unverifiedMtwAddress
+    pepperQuotaRemaining = distributedBlindedPepper
+    requestQuotaRemaining = requestSubsidisedAttestation
+
+    // No pepper quota remaining is only bad if it's not already cached. Given Komenci will fetch
+    // a pepper for you once, a session could be invalid due to the pepper condition if a user
+    // fetched their pepper once this session then uninstalled without starting a new session
+    if (
+      (!ownPepper && !distributedBlindedPepper) ||
+      !requestSubsidisedAttestation ||
+      !submitMetaTransaction
+    ) {
+      sessionActive = false
```

Comment:
We should log some kind of error here.

Reply:
Will put a logger.debug for now and make sure to instrument this with analytics when I have time to do a pass for that